packages/databyss-api/src/routes/api/helpers/pagesHelper.js

```diff
@@ -33,19 +37,37 @@ const getSourcesFromId = list => {
     const source = await Source.findOne({
       _id,
     }).catch(err => console.log(err))
+    if (!source) return {}
+
     const { resource } = source
     const response = { rawHtml: resource, _id }
     return response
   })
   return Promise.all(promises)
 }
 
+const getTopicsFromId = list => {
```

Comment:
This is almost a duplicate of getEntriesFromId. Let's refactor to a more generic form, like:
```
const populateRefEntities = (list, Model) =>
  Promise.all(list.map(async b => {
    const _id = b.refId
    const entity = await Model.findOne({ _id })
    if (!entity) { throw new BadRefIdError(refId) }
    return { rawHtml: entity.text }
  }))
```
Note that we should create a special error type `BadRefIdError` so that we can monitor these in the logs and fix data integrity problems.

Also note that this requires normalizing the entity models (Entry, Topic, Source, etc) to have a `text` key that has the text of the entity.

Reply:
`const entity = await eval(Model).findOne({ _id })`
gives me the error that Source, Entry, Topic are used but never defined when running tests